### DISCUSSION TITLE
Printing customized text

### DISCUSSION THREAD

**Discussion by shawna-slh** [w3c/low-vision-a11y-tf#107] [General]
**Summary:**
* User Need: Users can print content after customizing how the text is displayed.
* That is not supported in PDF. It is for HTML, with some limitations.
* [3.7.2 Printing Customized Text](https://w3c.github.io/low-vision-a11y-tf/requirements.html#printing-customized-text) in Users Needs / Requirements doc

**Previous drafts and discussions:**
* [Printing Customized Text](https://www.w3.org/WAI/GL/low-vision-a11y-tf/wiki/Printing_Customized_Text) LVTF wiki page
* [Printing Customized Text #76](https://github.com/w3c/wcag21/issues/76) WCAG 2.1 GitHub issues

**Note:**
* Fundamentally, this may be purely a user agent issue. However, if it works for content provided in some formats (HTML, docs) and not others (PDF), then it is currently an authoring issue.

**Related in WCAG 2:**
* <a href="https://www.w3.org/TR/WCAG22/#visual-presentation">1.4.8 Visual Presentation</a>
* <a href="https://www.w3.org/TR/WCAG22/#text-spacing">1.4.12 Text Spacing</a>

<hr>

**To Do:**
* Now that we are doing this as supplemental guidance, and not a proposed SC, review the previous discussions and figure out how to present it succinctly and clearly as a basic best practice, then what we need to note as exceptions.

**Comment by mraccess77** [comment]
This could fall into 2 groups:
1.  User agent doesn't allow/support customization of print size
2. Author settings block customization of print size

**Comment by Myndex** [comment]
"Print version of this page" is a long standing feature of some websites, to reduce the format of a dynamic web page to print well onto an A4 or 8.5x11 page. In that sense it is under author control, not user agent.

User agents though do usually have some features, namely "ignore background color when printing".

If the user changes the look or mode on screen, that is reflected in the print in at least the user agents I've tested. For instance, if you go into reader mode and print, you'll get the reader mode format on the hard copy.

_On the other hand,_ simply zooming text in or out does not reflect in the printed version.

### What the user can do now
A custom user style sheet with a media query that restricts it to print only, such as:

```
@media print { 
 /* All your print styles go here */
 
   /* Set Root element to a large print size font */
   :root {
 	font-size: 24pt !important; /* this is equivalent to 32px on screen */
 	margin: 0 !important;
 	padding 0 !important;
 	color: black !important; 
 	}
 
   body, section, article {
 	max-width: 100% !important;
 	font-size: 1rem !important;
 	margin: 0 !important;
 	padding 0 !important;
 	}
    header, footer, nav, #header, #footer, #nav { display: none !important; } 
}
```

While this sheet is set to print only, so it does not affect viewing the site on a monitor, it does not fully solve the print problem as every site responds differently due to different HTML semantics (not to mention version) and different CSS strategies.

Some sites use HTML5, some are still on 4. Some use bootstrap/columns, other use tables, other use some combination of div and class.... what works for one site fails for another, in terms of user style sheets, and even using ` !important;` does not seem to override always.

I suppose it might be possible to create a default sheet that cleared everything... but that is still at the mercy of the HTML structure, whcih for dynamic sites is now often a function of javascript...

**Comment by bruce-usab** [comment]
I just this morning saw my name in @mraccess77 lvtf email 1/4.  I will help wrestle with this one.

Sharing an anecdote, as my spouse is in the habit of printing web pages, consistent with motivations listed in resources linked from this page.   My own personal experience is that printing is often broken _without_ the user making any changes to the default web presentation!  (Text cut off, text too small, etc., consistent with the linked example problems.)

I think we might want to broaden the problem statement:  Print versions are important to accessibility.  It follows that print functionality is disproportionately important to PWD.  Being able to customize the print rendition is one piece of that.

One subtle barrier I was hoping we might highlight is when print-oriented documents (e.g., PDF, Word) are okay (WCAG 2x conforming) on-screen , but not sufficient as hard copy hand-outs.  Examples:  9 or 10pt body text; a paragraph of pure red text (poor, but manageable on-screen; nothing but pink fuzz in print).  Many employees with significant low vision can usually manage with the majority of paper handed out during a business meeting.  (Probably, they carry a loupe.  Plus they plan to can catch up using their desk-sized CCTV after the meeting.)  But then there will be that one document that gets most of the focus, like a bad photocopy, and then suddenly and unexpectedly the person cannot follow the conversation.
